File: trupti/python/wordle_api.py
import requests 
import json 

session = requests.Session()

register_url = "https://wordle.we4shakthi.in/game/register"
create_url = "https://wordle.we4shakthi.in/game/create"
guess_url = "https://wordle.we4shakthi.in/game/guess"

register_data = {"mode": "wordle","name": "string"}
response = session.post(register_url , json = register_data)
print("rgister status : " , response.status_code)
register_json = response.json()
print("register response :" , register_json)
game_id = str(register_json.get("id"))

create_data = {"id": game_id,"overwrite": True}
response = session.post(create_url , json = create_data)
print("create status : " , response.status_code)
print("create response : " , response.json())

guess_data = {"guess" : "apple" , "id" : game_id}
response = session.post(guess_url, json = guess_data)
print("guess status : " , response.status_code)
print("guess response : " , response.json())

File: trupti/scala/wordle_api.scala
// //> using dep "com.lihaoyi::requests:0.9.0"
// //> using dep "com.lihaoyi::ujson:4.2.1"

// import requests.Session
// import ujson.Obj

// var sessionCookie: String = ""
// var gameId: String = ""
// val baseUrl = "https://wordle.we4shakthi.in/game/"

// @main def main(): Unit = 
//   registerAndCreateGame()
//   makeGuess("APPLE")

// def registerAndCreateGame(): Unit = 
//   println("--- 1. Registering and Creating Game ---")
//   try {
//     val regPayload = Obj("mode" -> "wordle", "name" -> "Trupti_Scala_Bot")
//     val regResponse = requests.post(s"${baseUrl}register", data = regPayload)
//     println(s"Register Status: ${regResponse.statusCode}")

//     sessionCookie = regResponse.cookies("session").getValue
//     gameId = ujson.read(regResponse.text())("id").str
//     println(s"Game ID: $gameId")

//     val createPayload = Obj("id" -> gameId, "overwrite" -> true)
//     val createResponse = requests.post(
//       s"${baseUrl}create",
//       data = createPayload,
//       cookieValues = Map("session" -> sessionCookie)
//     )
//     println(s"Create Status: ${createResponse.statusCode}")

//     // The session cookie might be updated upon game creation.
//     if (createResponse.cookies.contains("session")) {
//       sessionCookie = createResponse.cookies("session").getValue
//     }
//     println("Game registered and created successfully.")
//   } catch {
//     case e: Exception =>
//       println(s"An error occurred during setup: ${e.getMessage}")
//   }


// def makeGuess(guessWord: String): Unit = 
//   println(s"\n--- 2. Making a Guess ---")
//   if (gameId.isEmpty || sessionCookie.isEmpty) {
//     println("Cannot make a guess. Please register a game first.")
//     return
//   }

//   println(s"Guessing word: $guessWord")
//   try {
//     val guessPayload = Obj("guess" -> guessWord.toLowerCase, "id" -> gameId)
//     val guessResponse = requests.post(
//       s"${baseUrl}guess",
//       data = guessPayload,
//       cookieValues = Map("session" -> sessionCookie)
//     )

//     println(s"Guess Status: ${guessResponse.statusCode}")
//     val guessJson = ujson.read(guessResponse.text())
//     println(s"API Response: $guessJson")
//   } catch {
//     case e: Exception =>
//       println(s"An error occurred during the guess: ${e.getMessage}")
//   }
